server/neptune/workflows/internal/lyft/audit_workflow_runner_wrapper.go

```diff
@@ -0,0 +1,62 @@
+package lyft
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/activities"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/config/logger"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deploy/revision/queue"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deploy/terraform"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/job"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/root"
+	"go.temporal.io/sdk/workflow"
+)
+
+type auditActivities interface {
+	AuditJob(ctx context.Context, request activities.AuditJobRequest) error
+}
+
+type AuditWorkflowRunnerWrapper struct {
+	Activity auditActivities
+	queue.TerraformWorkflowRunner
+}
+
+func (w *AuditWorkflowRunnerWrapper) Run(ctx workflow.Context, deploymentInfo terraform.DeploymentInfo) error {
+	if err := w.emit(ctx, job.Running, deploymentInfo); err != nil {
+		return errors.Wrap(err, "emitting atlantis job event")
+	}
+
+	if tfRunErr := w.TerraformWorkflowRunner.Run(ctx, deploymentInfo); tfRunErr != nil {
```

Comment:
workflows won't fail if the terraform run fails so this is incorrect. You need to emit when we receive the signal on the state change.

Reply:
Ahh right! I've changed the implementation to emit events on state change!